feat: add cloud secrets manager docs (#537)
kestra-io/docs: content/docs/09.administrator-guide/01.configuration/03.enterprise-edition/secrets/aws-secret-manager/index.md

---
title: AWS Secret Manager configuration
---

This integration will store secrets in [AWS Secret Manager](https://aws.amazon.com/secrets-manager/). Kestra authentications must have `CreateSecret, DeleteSecret, DescribeSecret, GetSecretValue, ListSecrets, PutSecretValue, RestoreSecret, TagResource, UpdateSecret` permissions ([reference](https://docs.aws.amazon.com/secretsmanager/latest/userguide/reference_iam-permissions.html)).

> ## Authentication
> You can handle authentication to AWS Cloud in multiple ways:
> - By secret using properties `accessKeyId` and `secretKeyId`.
> - Using a session using the `sessionToken`,  `accessKeyId` and `secretKeyId` properties.
> - If none is set, the default authentication will be used like for AWS own cli.

Reviewer: ```suggestion
## Authentication

You can configure the authentication to AWS Cloud in multiple ways:
- Using `accessKeyId` and `secretKeyId` properties
- Adding `sessionToken`,  `accessKeyId` and `secretKeyId` properties
- If the above properties are not set, Kestra will use the default AWS authentication, in the same way as AWS CLI handles it (i.e. trying to use the AWS CLI profile or the default environment variables `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_DEFAULT_REGION`)
```

```yaml
kestra:
  secret:
    aws-secret-manager:
      accessKeyId: "test"
      secretKeyId: "test"
      sessionToken: "token"
```

> ## Others configurations

Reviewer: ```suggestion
## Additional configuration
```

### `kestra.secret.aws-secret-manager.region`

> AWS region to use.

Reviewer: ```suggestion
The AWS region to be used by the Secrets Manager
```

### `kestra.secret.aws-secret-manager.prefix`

> Optional. All key handled by kestra will be prefixed with this. Can be useful to share the same secret manager between Kestra instances.

Reviewer: ```suggestion
Optional property to store secrets separately for a different namespace, tenant, or instance. If configured, Kestra will prefix all Secret keys using that prefix. The main purpose of a prefix is to share the same secret manager between multiple Kestra instances.
```

### `kestra.secret.aws-secret-manager.endpointOverride`

> Optional. Replace AWS default endpoint.

Reviewer: ```suggestion
Optional property to replace AWS default endpoint by an AWS-compatible service such as [MinIO](https://min.io/).
```
